### PR TITLE
[FIX] mrp: allow MOs without components

### PR DESCRIPTION
**Pull request by diagnoza**
After MRP refactoring, all newly created MOs originating from the split (done [here](https://github.com/odoo/upgrade/blob/master/migrations/mrp/saas~13.4.2.0/end-migrate.py#L107)) have no `move_raw_ids` and yet should be `done` instead of `draft`. Part of a unicorn task [opw-3100354](https://www.odoo.com/web#id=3100354&cids=1&model=project.task&view_type=form). This PR is a backport of an already existing solution in 16.0: https://github.com/odoo/odoo/commit/bf5e1debf9c684c1bbf3ca440a87001fd7faf9fa.

Current behavior before PR:
Many MOs created during the upgrade (resulting from the mentioned split) are in state `draft`.

Desired behavior after PR is merged:
MOs created as a result of the split are in state `done`.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -1511,9 +1514,8 @@ def _generate_backorder_productions(self, close_mo=True):
         if not close_mo:
             self.move_raw_ids.filtered(lambda m: not m.additional)._do_unreserve()
             self.move_raw_ids.filtered(lambda m: not m.additional)._action_assign()
-        # Confirm only productions with remaining components
-        backorders.filtered(lambda mo: mo.move_raw_ids).action_confirm()
-        backorders.filtered(lambda mo: mo.move_raw_ids).action_assign()
+        backorders.action_confirm()
+        backorders.action_assign()
```
Comment (ticodoo) [inline addons/mrp/models/mrp_production.py:1518]: Also seems like an unnecessary change for v14/the issue

File: addons/mrp/models/stock_rule.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def _run_manufacture(self, procurements):
             self.env['stock.move'].sudo().create(productions._get_moves_raw_values())
             self.env['stock.move'].sudo().create(productions._get_moves_finished_values())
             productions._create_workorder()
-            productions.filtered(lambda p: not p.orderpoint_id and p.move_raw_ids).action_confirm()
+            productions.action_confirm()
```
Comment (ticodoo) [inline addons/mrp/models/stock_rule.py:60]: Unrelated to your issue

File: addons/mrp/views/mrp_production_views.xml [hunk 3]
```diff
@@ -63,15 +63,29 @@
...
+		    <button name="button_mark_done" attrs="{'invisible': [
+                        '|',
+                        ('confirm_no_consumption', '=', False),
                         '&amp;',
                         '|',
                         ('state', 'not in', ('confirmed', 'progress')),
                         ('qty_producing', '!=', 0),
-                        ('state', '!=', 'to_close')]}" string="Mark as Done" type="object" class="oe_highlight"/>
+                        ('state', '!=', 'to_close')]}" string="Mark as Done" type="object" class="oe_highlight"
+                            confirm="There are no components to consume. Are you still sure you want to continue?"/>
```
Comment (ticodoo) [inline addons/mrp/views/mrp_production_views.xml:88]: Would need to be updated to match: https://github.com/odoo/odoo/pull/75827/files from previous comment

**Review by ticodoo — CHANGES_REQUESTED** [review]
Hello! Is there no way to fix this in the upgrade script? (disclaimer: I didn't work on the script and don't know it super well). E.g. make it so the MOs created by the split are manually updated to be "done" if their original MO was also done after the compute has been triggered?

Ideally since this is only an upgrade issue, we would only update the script since this change is risky for stable + it will change the current behavior for users. Also as you can see from my many comments, this change is a bit of a pain to adapt for v14.

**Comment by diagnoza** [comment]
> Hello! Is there no way to fix this in the upgrade script? (disclaimer: I didn't work on the script and don't know it super well). E.g. make it so the MOs created by the split are manually updated to be "done" if their original MO was also done after the compute has been triggered?
> 
> Ideally since this is only an upgrade issue, we would only update the script since this change is risky for stable + it will change the current behavior for users. Also as you can see from my many comments, this change is a bit of a pain to adapt for v14.

First off, I greatly appreciate your thorough review @ticodoo, thanks a lot.

I thought I'd make it an upgrade script, but the Accounting Team advised me to fix it in standard. Besides, even if we manually set the state during the upgrade, there's nothing preventing it from being recomputed in the future (when a MO is opened or during future upgrades for instance). Do you still think it's preferable to have it set manually from an upgrade script or should I continue porting this solution?
